### PR TITLE
Chore: codeownership for feature toggles

### PR DESCRIPTION
**Pull request by ArturWierzbicki**
Part of https://github.com/grafana/grafana-enterprise/issues/4731

This PR:
1. Changes the name of `FeatureFlag` struct to `FeatureToggle`
2. Sorts the toggles alphabetically to avoid further changes to the generated files once more squads extract their toggles from the registry file
3. Extracts toggles owned by the multitenancy squad to a separate file

### CODE REVIEW

File: packages/grafana-data/src/types/featureToggles.gen.ts [hunk 1]
```diff
@@ -16,68 +16,68 @@
...
-  authnService?: boolean;
-  disablePrometheusExemplarSampling?: boolean;
-  alertingBacktesting?: boolean;
-  editPanelCSVDragAndDrop?: boolean;
-  alertingNoNormalState?: boolean;
-  logsSampleInExplore?: boolean;
-  logsContextDatasourceUi?: boolean;
-  lokiQuerySplitting?: boolean;
-  individualCookiePreferences?: boolean;
-  drawerDataSourcePicker?: boolean;
```
Comment (nmarrs) [inline packages/grafana-data/src/types/featureToggles.gen.ts:82]: So many feature flags 😅

File: pkg/services/featuremgmt/registry/multitenancy_squad.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+			RequiresDevMode: true,
+		},
+		{
+			Name:            "queryLibrary",
+			Description:     "Reusable query library",
+			State:           FeatureStateAlpha,
+			RequiresDevMode: true,
+		},
+	}
+)
```
Comment (nmarrs) [inline pkg/services/featuremgmt/registry/multitenancy_squad.go:83]: Is this the expected format for teams moving forward? I.e for each team to split out their feature toggles into a separate file?

**Review by nmarrs — APPROVED** [review]
LGTM

File: pkg/services/featuremgmt/registry/multitenancy_squad.go (hunk 2 above)
Comment (ArturWierzbicki) [reply pkg/services/featuremgmt/registry/multitenancy_squad.go:83]: yes - I added a comment here https://github.com/grafana/grafana/blob/29a827d65006ec609d4dd83e866f930f82887fd5/pkg/services/featuremgmt/registry.go#L19-L23

File: packages/grafana-data/src/types/featureToggles.gen.ts (hunk 1 above)
Comment (ArturWierzbicki) [reply packages/grafana-data/src/types/featureToggles.gen.ts:82]: 😢

File: pkg/services/featuremgmt/registry/feature_toggle.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
+package featuremgmt_registry
```
Comment (ryantxu) [inline pkg/services/featuremgmt/registry/feature_toggle.go:1]: can we use a standard package name matching the folder?

Comment (ArturWierzbicki) [reply pkg/services/featuremgmt/registry/feature_toggle.go]: 👍  done

**Review by ryantxu — APPROVED** [review]

File: pkg/services/featuremgmt/registry_backend_platform_squad.go [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package featuremgmt
```
Comment (mildwonkey) [inline pkg/services/featuremgmt/registry_backend_platform_squad.go:1]: I'm not comfortable codifying conway's law quite this precisely; let's not literally order our codebase by the organizational structure & team names. These toggle files can be grouped similarly, but with more loose grouping, by service or domain (dashboards, navigation, etc) in a way that matches the existing code layout.

**Review by mildwonkey — COMMENTED** [review]
I have some strong opinions about those registry file names, but I'm good with the general concepts.

File: pkg/services/featuremgmt/registry_backend_platform_squad.go (hunk 4 above)
Comment (ryantxu) [reply pkg/services/featuremgmt/registry_backend_platform_squad.go:1]: Sounds good -- from @ArturWierzbicki design doc, it sounds like the main goal was wanting a valid code owner for every toggle.  The actual file names could be anything else.

Comment (ArturWierzbicki) [reply pkg/services/featuremgmt/registry_backend_platform_squad.go:1]: That sounds good, but I am not sure how to actually do that in practice. Should we just replicate the service folders? How would we handle situations in which team A adds feature toggle to a domain owned by team B (e.g. public dashboards and dashboards before the team handover)?

Comment (ArturWierzbicki) [reply pkg/services/featuremgmt/registry_backend_platform_squad.go:1]: or - we could also move each registry file to the corresponding service package.. but that would result in cyclic dependencies

**Review by chri2547 — APPROVED** [review]

File: pkg/services/featuremgmt/registry_backend_platform_squad.go (hunk 4 above)
Comment (ryantxu) [reply pkg/services/featuremgmt/registry_backend_platform_squad.go:1]: And many are frontend toggles that do not map to the backend service package layouts

Comment (ArturWierzbicki) [reply pkg/services/featuremgmt/registry_backend_platform_squad.go:1]: hey @mildwonkey @tolzhabayev, what do you think?

Comment (mildwonkey) [reply pkg/services/featuremgmt/registry_backend_platform_squad.go:1]: @ArturWierzbicki as a rough heuristic that's a good start. Use relevant service names as the file name where possible: dashboards, alerts, etc. If you  need to split out a single group due to multiple codeowners, use your best judgement: if we need a single flag in a file, go for it; if they can be grouped in another way that still makes sense outside of our organizational structure (for e.g., a set of flags for the same overall feature, even if the flags themselves span services), go with that. Split frontend vs backend flags if necessary for codeowners, too. ( {featurewithsimilarflag}_frontend.go, {featurewithsimilarflag}_service.go, something like that)

**Comment by ArturWierzbicki** [comment]
closing in favor of https://github.com/grafana/grafana/pull/64266

**Pull request closed without merging**